packages/scaffold-ui/src/partials/w3m-connecting-wc-mobile/index.ts

```diff
@@ -36,11 +41,20 @@ export class W3mConnectingWcMobile extends W3mConnectingWidget {
     if (this.wallet?.mobile_link && this.uri) {
       try {
         this.error = false
-        const { mobile_link, name } = this.wallet
-        const { redirect, href } = CoreHelperUtil.formatNativeUrl(mobile_link, this.uri)
+        const { mobile_link: deep_link, link_mode: universal_link, name } = this.wallet
+        let link = deep_link
+        let target: '_self' | '_blank' = '_self'
+        if (OptionsController.state.enableUniversalLinks) {
+          if (!universal_link) {
+            throw new Error('Universal Link is undefined.')
+          }
```

Comment:
emabling universal links makes  it so it's required to have one? Or should it use one if it has it and fall back to previous behavior if not?

Reply:
Aren't you calling `filterWAlletsWithUniversalLinks` ? You could make it so it's types as returning string[] and that would sort it out I think 🤔

Reply:
that function is only call if the flag is set to true, so it's not possible to determinate the right type there